src/server/system_services/replication_store.js

```diff
@@ -86,22 +86,69 @@ class ReplicationStore {
+        dbg.log1('update_log_replication_status_by_id: ', _id, last_cycle_end);
+        const parsed_id = db_client.instance().parse_object_id(_id);
+        const find = { _id: parsed_id};
+        const update = { $set: { 'log_replication_info.status.last_cycle_end': last_cycle_end } };
+        const options = { returnOriginal: false };
+        dbg.log1('update_log_replication_status_by_id: ', find, update, options);
+        const ans = await this._replicationconfigs.findOneAndUpdate(find, update, options);
+        dbg.log1('update_log_replication_status_by_id: ans', ans);
+        this._replicationconfigs.validate(ans.value, 'warn');
+        return ans;
+    }
+
+    async update_log_replication_marker_by_id(_id, rule_id, log_marker) {
+        dbg.log1('update_log_replication_marker_by_id: ', _id, rule_id, log_marker);
+        const parsed_id = db_client.instance().parse_object_id(_id);
+        const find = { _id: parsed_id, 'rules.rule_id': rule_id };
+
+        const updates = {
+            'rules.$.rule_log_status': {
+                log_marker
+            }
+        };
+        const update = { $set: updates };
+        const options = { returnOriginal: false };
+        dbg.log1('update_log_replication_marker_by_id: ', find, update, options);
+        const ans = await this._replicationconfigs.findOneAndUpdate(find, update, options);
+        dbg.log1('update_log_replication_marker_by_id: ans', ans);
+        this._replicationconfigs.validate(ans.value, 'warn');
+        return ans;
+    }
+
     async find_rules_updated_longest_time_ago() {
-        dbg.log0('find_rules_updated_longest_time_ago: ');
+        dbg.log1('find_rules_updated_longest_time_ago: ');
         // TODO: postgres client does not support $project with $min op,
         // for doing an in db operation we will need to add support 
         // for aggregation framework in postgres client 
         const replications = await this._replicationconfigs.find({ deleted: null });
-        dbg.log0('find_rules_updated_longest_time_ago: ', replications);
+        dbg.log1('find_rules_updated_longest_time_ago: ', replications);
 
         const reduced_replications = _.map(replications, repl => ({
             replication_id: repl._id,
             rule: _.minBy(repl.rules, rule => (rule.rule_status && rule.rule_status.last_cycle_end) || 0) //least_recently_replicated_rule
         }));
-        dbg.log0('find_rules_updated_longest_time_ago: ', reduced_replications);
+        dbg.log1('find_rules_updated_longest_time_ago: ', reduced_replications);
 
         return reduced_replications;
     }
+
+    async find_aws_log_based_replication_rules() {
+        dbg.log1('find_aws_log_based_replication_rules: ');
+        const replications = await this._replicationconfigs.find({ deleted: null });
+
+        const aws_reduced_replications = _.filter(replications, repl => repl.log_replication_info.aws_log_replication_info);
+        // TODO: Debug and fix
```

Comment:
Why leave the code in the comment?
Are there cases where we will use it?

Reply:
I think this came from Kfir's code. I don't know how relevant this comment is to others however it communicated to me that there is further filtering that can be done here in order to avoid multiple `ifs` in the rest of the code. Having said that, I don't know how buggy this commented code is (or if it is buggy at all).

Reply:
If we don't use it or do not know when we will use it in the future, it is better to delete it.

Reply:
Marking resolved - addressed.